iso15118/shared/security.py

```diff
@@ -1376,21 +1373,29 @@ def get_certificate_hash_data(
         return None
 
     all_certificates = all_certificates_from_chain(certificate_chain, root_cert)
-    # Each certificate is followed by its issuer, except for the root,
+    # The `all_certificates` list will have the following line-up
+    # [leaf, subca2, subca1, root]
+    # Thus, each certificate is followed by its issuer, except for the root,
     # which is self-signed.
-    certificate_and_issuer_pairs = [
-        (all_certificates[i], all_certificates[i + 1])
-        for i in range(len(all_certificates) - 1)
-    ] + [(root_cert, root_cert)]
-
-    return [
-        derive_certificate_hash_data(certificate, issuer)
-        for certificate, issuer in certificate_and_issuer_pairs
-    ]
+    hash_data: List[Dict[str, str]] = []
+    for idx, certificate in enumerate(all_certificates):
+        try:
+            if idx < len(all_certificates) - 1:
+                hash_data.append(
+                    derive_certificate_hash_data(certificate, all_certificates[idx + 1])
+                )
+            else:
+                # the last entry of the list contains the root_cert which
+                # is a self-signed certificate
+                hash_data.append(derive_certificate_hash_data(root_cert, root_cert))
+        except (ExtensionNotFound, OCSPServerNotFoundError):
+            # if we cant extract the OCSP data, then we ignore it and move on
```

Comment:
Does this help anything?  If one of the sub-CA certificates is missing its responder URL, this will assemble the rest of the chain, but it won't be a complete one.  Wouldn't it just fail anyway later on?

Reply:
I was wondering what should be the best... we may have enough data to request the OCSP result for some of the certificates but not all of them, so shall we do it or shall we just completely drop this action and do not do any OCSP call?

Reply:
I reflected about it and you are right, there is no point on returning an incomplete hash data, it is all or nothing